src/Cryptol/TypeCheck/SimpType.hs

```diff
@@ -113,6 +113,7 @@ tSub :: Type -> Type -> Type
 tSub x y
   | Just t <- tOp TCSub (op2 nSub) [x,y] = t
   | tIsInf y  = tError (tf2 TCSub x y)
+  | tIsInf x = x
```

Comment:
Should this require that `y` be finite? (I'm thinking of what happens if you have `inf - inf`.)

Reply:
I think that we should have already asserted that `y` is finite, it is part of the type of `-` (have a look in `Cryptol.cry`).  Also when we make `tSub` terms we should take care to preserve this invariant, although I think we have another tickets where this invariant was violated by a rewrite rule.

Reply:
Ah, good point. Can you add a comment to `tSub` saying that it assumes the precondition that the second argument is finite? (It might also be worth adding similar comments to other functions in this module that assume preconditions, such as `tDiv` and `tMod`.)